config/routes.rb

```diff
@@ -1,11 +1,13 @@
-Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+Rails.application.routes.draw do  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   require 'sidekiq/web'
   require 'sidekiq/cron/web'
   mount Sidekiq::Web => '/sidekiq'
   resources :users do
     resources :products
   end
+  get '/products', to: 'products#index'
```

Comment:
This looks pretty standard; why not use `resources`, with a whitelist of actions that you need?